.github/workflows/swiftlint.yml

```diff
@@ -0,0 +1,16 @@
+name: SwiftLint
+on:
+  workflow_dispatch: {}
+  pull_request:
+    paths:
+      - '.github/workflows/swiftlint.yml'
+      - '.swiftlint.yml'
+      - '**/*.swift'
```

Comment:
This is general way of linting swift projects. It will run on any PR where the mentioned path are changed.

- If Ci config changed
- if lint YML file changed 
- If any swift file changed inside the project

Reply:
I did this in one of my [old projects](https://github.com/mikaelacaron/brain-marks/blob/dev/.github/workflows/main.yml) and just mentioned the branch name, is there a benefit to specifically listing files, vs just the branch name?

Reply:
@rizwankce I'm still going to merge this cause I want to see the result, but I'd like still to know the answer to that question, if you know it, thanks!

Reply:
After merging I had to change the script, it does look like it was only working when manually clicking the button, and not on a PR, with Swift files changed.
I did also change it do run when commits are pushed to `main` or `dev`

Reply:
Oh wow. Lots of things happened while I sleep. I'm from the other side of the world. 

To answer your question, if we target based on branch name. Job will run even for documentation changes or any other settings changes. That's why we only prefer to run when config files changes or any swift file changes. We are not gonna lint any other files with swift lint. Generally Mac OS machines are costly in CI. So typically we try to save the extra job which is going to be wasted. Other than that I don't find any other reason. If it works we can keep it as it is.

Reply:
haha the 1st week of Hacktoberfest is always the busiest!
That's a good point! I'll keep it like this for right now, because it only runs on PRs (when I approve of it) so it should be fine if I start hitting limits, I'll change it back to the `path`
Thanks for the insight!

Reply:
You won't be hitting any limits. GitHub actions for OSS projects is unlimited. Even for Mac OS.

Reply:
Oh! I thought it was still capped, good to know!! Thanks!